### DISCUSSION TITLE
How to track result with type utility?

### DISCUSSION THREAD

**Discussion by tetthys** [prisma/prisma#23166] [Q&A]
### Question

Is there any type utility like

```Prisma.Result<'user', 'findFirst'>```

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by nurul3101** [comment]
Hello @tetthys 👋 

Can you please elaborate on your question?
I didn't quite get what you are trying to achieve.

**Reply by tetthys** [reply]
I'm using dynamic table like

```
await (this.model as any).findFirst(args);
```

But it loses this type of result.

**Reply by nurul3101** [reply]
Do you want to pass the table name dynamically to a PrismaClient query?

**Reply by tetthys** [reply]
Yes, I created a custom wrapper class corresponding to each model.

**Reply by ludralph** [reply]
Hi @tetthys 

Yes, there is a type utility in Prisma similar to what you want. Prisma provides a `Result` type utility that can be used to infer the result type of a specific operation on a model. See the [documentation](https://www.prisma.io/docs/orm/prisma-client/client-extensions/type-utilities#type-utilities).

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!;
